pkg/capi/gcp.go

```diff
@@ -0,0 +1,294 @@
+			Skip("Skipping GCP E2E tests")
+		}
+		oc, _ := framework.NewCLI()
+		framework.SkipIfNotTechPreviewNoUpgrade(oc, cl)
+
+		infra, err := framework.GetInfrastructure(ctx, cl)
+		Expect(err).NotTo(HaveOccurred(), "Failed to get cluster infrastructure object")
+		Expect(infra.Status.InfrastructureName).ShouldNot(BeEmpty(), "infrastructure name was empty on Infrastructure.Status.")
+		clusterName = infra.Status.InfrastructureName
+
+		framework.CreateCoreCluster(ctx, cl, clusterName, "GCPCluster")
+		mapiMachineSpec = getGCPMAPIProviderSpec(cl)
+	})
+	AfterEach(func() {
+		if platform != configv1.GCPPlatformType {
+			// Because AfterEach always runs, even when tests are skipped, we have to
+			// explicitly skip it here for other platforms.
+			Skip("Skipping GCP E2E tests")
+		}
+		framework.DeleteCAPIMachineSets(ctx, cl, machineSet)
+		framework.WaitForCAPIMachineSetsDeleted(ctx, cl, machineSet)
+		framework.DeleteObjects(ctx, cl, gcpMachineTemplate)
+	})
+	// OCP-77825 capi created instances support disks pd-ssd and pd-standard
+	// author: miyadav@redhat.com
+	DescribeTable("should be able to run a machine with disk types",
+		func(diskTypeName string, expectedDiskType gcpv1.DiskType) {
+			mapiProviderSpec := getGCPMAPIProviderSpec(cl)
+			Expect(mapiProviderSpec).ToNot(BeNil())
+			mapiProviderSpec.OnHostMaintenance = "Migrate"
+			mapiProviderSpec.ConfidentialCompute = "Disabled"
+			mapiProviderSpec.Disks[0].Type = diskTypeName
+			gcpMachineTemplate = createGCPMachineTemplate(cl, mapiProviderSpec)
+			machineSet, _ = framework.CreateCAPIMachineSet(ctx, cl, framework.NewCAPIMachineSetParams(
+				"gcp-machineset-77825",
+				clusterName,
+				mapiMachineSpec.Zone,
+				1,
+				corev1.ObjectReference{
+					Kind:       "GCPMachineTemplate",
+					APIVersion: infraAPIVersion,
+					Name:       gcpMachineTemplateName,
+				},
+			))
+			Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+			framework.WaitForCAPIMachinesRunning(framework.GetContext(), cl, machineSet.Name)
+		},
+		Entry("Disk type pd-standard", "pd-standard", gcpv1.PdStandardDiskType),
+		Entry("Disk type pd-ssd", "pd-ssd", gcpv1.PdSsdDiskType),
+	)
+	// OCP-74795 - add support for shielded VMs - It takes defaults if configs are not supported; eg-vtpm alone set to Enabled will result in IntergrityMonitoring also as Enabled
+	// doesn't matter what we pass - all enabled/secureboot only disabled(default)/secureboot only enabled are valid changes which we can apply
+	// author: miyadav@redhat.com
+	DescribeTable("should configure Shielded VM options correctly",
+		func(enableSecureBoot mapiv1.SecureBootPolicy, enableVtpm mapiv1.VirtualizedTrustedPlatformModulePolicy, enableIntegrityMonitoring mapiv1.IntegrityMonitoringPolicy) {
+			mapiProviderSpec := getGCPMAPIProviderSpec(cl)
+			Expect(mapiProviderSpec).ToNot(BeNil())
+			mapiProviderSpec.OnHostMaintenance = "Migrate"
+			mapiProviderSpec.ConfidentialCompute = "Disabled"
```

Comment:
@miyadav I still think we shouldn't add these parameters in the gcpmachinetemplate, which will cause each case to need to assign values ​​to these parameters like you do here. What I want is that we create the default and simplest gcpmachinetemplate, and then we assign values ​​to the parameters which we want to test. like below:
```
		gcpMachineTemplate = createGCPMachineTemplate(cl, mapiProviderSpec)
		gcpMachineTemplate.Spec.Template.Spec.OnHostMaintenance =(*gcpv1.HostMaintenancePolicy)(&mapiProviderSpec.OnHostMaintenance)
```